### ISSUE TITLE
Integrate ELECTRA Pretrained Model into Keras_NLP Ecosystem

### ISSUE THREAD

**Issue by pranavvp16** [keras-team/keras-nlp#1281]
I would like to add ELECTRA pretrained model in the keras_nlp ecosystem. I have went through the `CONTRIBUTINT_MODELS.md` and making an issue is the first task to add the model. I have pretty good understanding of transformers and have went through the ELECTRA architecture mentioned [here](https://github.com/google-research/electra).  So if there is no problem regarding the integration I can start working on this !!

**Comment by mattdangerw** [comment]
Electra would be a welcome addition! Thanks! As mentioned in the model contribution guide, first thing to do would be to write a backbone and a checkpoint conversion script from a known source that shows we can match outputs.

Huggingface might be the easiest checkpoint source. [The original electra repo](https://github.com/google-research/electra) would be totally fine as well, but looks like the implementation is in tf1, which would be a little painful to work with.

**Comment by pranavvp16** [comment]
Thanks for assigning the issue, I'll start working on it

**Comment by shivance** [comment]
@mattdangerw I had one Colab floating around which would implement electra in KerasNLP, will share if I find

**Comment by shivance** [comment]
#794

**Comment by mattdangerw** [comment]
> I'm trying to implement the backbone of ELECTRA which is similar to BERT with some changes. Please let me know if I'm wrong here cause I'm pretty new to this thing. I have implemented the backbone successfully in keras according to the ELECTRA architecture, and working of weight conversion from hugging face checkpoint

Yes that's definitely a good approach! Let's start with the backbone, not the pretraining approach.

The pretraining approach would make for an excellent keras.io example (which was https://github.com/keras-team/keras-nlp/issues/794), but beginning with a backbone is the right first step on adding this model to our API.

Will take a look at the issues on the PR shortly!

**Comment by dhantule** [comment]
Hi @pranavvp16, could you please confirm if the issue is resolved ?

**Comment by pranavvp16** [comment]
@mattdangerw I think this done long ago we can close this

**Comment by dhantule** [comment]
Hi @pranavvp16, closing this, feel free to open another issue.
